services/repository.py

```diff
@@ -27,36 +28,67 @@
 from services.bots import (
     get_adapter_auth_information,
 )
+from services.bots.github_apps import (
+    get_github_app_token,
+    get_specific_github_app_details,
+)
+from services.github import get_github_app_for_commit
 from services.yaml import read_yaml_field
 
 log = logging.getLogger(__name__)
 
 merged_pull = re.compile(r".*Merged in [^\s]+ \(pull request \#(\d+)\).*").match
 
 
-def _is_repo_using_integration(repo: Repository) -> bool:
-    owner = repo.owner
-    default_ghapp_installation = list(
-        filter(
-            lambda obj: obj.name == GITHUB_APP_INSTALLATION_DEFAULT_NAME,
-            owner.github_app_installations or [],
-        )
+def get_repo_provider_service_for_specific_commit(
+    commit: Commit,
+    fallback_installation_name: str = GITHUB_APP_INSTALLATION_DEFAULT_NAME,
+) -> torngit.base.TorngitBaseAdapter:
+    """Gets a Torngit adapter (potentially) using a specific github app as the authentication source.
+    If the commit doesn't have a particular app assigned to it, return regular `get_repo_provider_service` choice
+
+    This is done specifically after emitting checks for a PR using GitHub apps, because only the app
+    that posted the check can edit it later on. The "app for a commit" info is saved in Redis by the NotifyTask.
+    """
+    repository = commit.repository
+    installation_for_commit = get_github_app_for_commit(commit)
+    if installation_for_commit is None:
+        return get_repo_provider_service(repository, fallback_installation_name)
+
+    ghapp_details = get_specific_github_app_details(
```

Comment:
We do a check in `get_repo_provider_service` to ensure that the repo is a GitHub repo. Should we do a check here as well? This will save us a database read to the GitHub App Installation table I think. 

In fact, many parts of this function is GitHub specific... can we skip some more steps below if it's not a GitHub commit?

Reply:
This check is in [get_github_app_for_commit](https://github.com/codecov/worker/blob/b8dcb6f4cb010c5c370ef2e9f40dd74e8e009183/services/github.py#L67).

From the docstring: "If the commit doesn't have a particular app assigned to it, return regular `get_repo_provider_service` choice".
Because apps only exist for GitHub it follows that only if it's a GitHub repo the code would get to this point.